pkg/asset/store/store.go

```diff
@@ -125,9 +127,27 @@ func (s *storeImpl) DestroyState() error {
 	return nil
 }
 
+// validateAssets checks for expired certificates and removes the expired certificates from
+// the installer state file.
+func validateAssets(assets map[string]json.RawMessage) error {
+	for assetName, data := range assets {
+		var assetExpirer tls.AssetExpirer
+		err := json.Unmarshal(data, &assetExpirer)
+		if err != nil {
+			logrus.Warnf("failed to unmarshal asset %s from state file: %s", assetName, err.Error())
+			continue
+		}
```

Comment:
My only concern is that some entry, for example, `*rhcos.Release` might throw the following warning.

```
 WARNING failed to unmarshal asset *rhcos.Release from state file: json: cannot unmarshal string into Go value of type tls.AssetExpirer 
```

Would it send any false alarms to customers/users? That is whether we should only check cert assets only?

Reply:
Correct. We should try to unmarshal only cert assets or when we cannot unmarshal, we should just continue without a warning.

Reply:
@tthvo @sadasu We wont necessarily know which assets are certs or contain cert/keys until they are unmarshalled. I am fine making a change here but it would be after the unmarshalling. We can't just assume that Cert  or Key is in the name because that pattern doesn't 100% follow. I can say only run the actual certificate validation if the cert and key exist though.